### ISSUE TITLE
Problem of TypeResolver for Interface

### ISSUE THREAD

**Issue by gevening** [Real-Serious-Games/Unity-Weld#27]
Hi,

In GetPublicMethods function, when the target is an interface, it will not return any MethodInfo.

`
[Binding]
public interface A
{
[Binding]
void TestFunc();
}
`

It is ok the change the following line from

`
return (new[] { type })
                .Concat(type.GetInterfaces())
                .SelectMany(i => i.GetMethods(BindingFlags.Public | BindingFlags.Instance));
`

to 

`
return (new[] { type })
                .Concat(type.GetInterfaces())
                .SelectMany(i => i.GetMethods());
`

Or what is the correct way to do the reflection on interface method. Thanks.

**Comment by RoryDungan** [comment]
So I just tested this and it looks like the current version of GetPublicMethods does return methods on interfaces. I did test making the change you suggested and it still works so I guess we could simplify it but I couldn't reproduce the issue you mentioned. 

Here's the code I used to test in [RoslynPad](https://roslynpad.net/):
```cs
interface ITest
{
    string GetValue();
}

var type = typeof(ITest);
var methods = (new[] { type })
    .Concat(type.GetInterfaces())
    .SelectMany(i => i.GetMethods(BindingFlags.Public | BindingFlags.Instance));

foreach (var method in methods)
{
    Console.WriteLine(method.Name);
}
```

In my testing, both `GetMethods(BindingFlags.Public | BindingFlags.Instance)` and `GetMethods()` return all methods for interfaces.
